### ISSUE TITLE
Bug(zod/no-empty-custom-schema): Optional breaks the rule

### ISSUE THREAD

**Issue by jtiala** [marcalexiei/eslint-plugin-zod#237]
### 🧩 Description

When chaining `.optional()` to `.custom()`, the rule reports an error. Notice, that inferring the type out of the schema still works, meaning this should be a valid use case for the function.

### 🪜 How to reproduce

```ts
import { z } from "zod";

// Works

export const test1Schema = z.custom<string>(
  (val) => {
    return typeof val === "string";
  },
  {
    error: "Error",
  },
);

export type Test1 = z.infer<typeof test1Schema>;

export const test2Schema = z.custom<string>((val) => typeof val === "string", {
  error: "Error",
});

export type Test2 = z.infer<typeof test2Schema>;

export const test3Schema = z.custom((val) => typeof val === "string", {
  error: "Error",
});

export type Test3 = z.infer<typeof test3Schema>;

// Fails

export const test4Schema = z
  .custom<string>(
    (val) => {
      return typeof val === "string";
    },
    {
      error: "Error",
    },
  )
  .optional();

export type Test4 = z.infer<typeof test4Schema>;

export const test5Schema = z
  .custom<string>((val) => typeof val === "string", {
    error: "Error",
  })
  .optional();

export type Test5 = z.infer<typeof test5Schema>;

export const test6Schema = z
  .custom((val) => typeof val === "string", {
    error: "Error",
  })
  .optional();

export type Test6 = z.infer<typeof test6Schema>;
```

### ✅ Expected behavior

I would expect all 6 test cases to be valid

### 💻 System info

```shell
Binaries:
    Node: 24.14.1 - /Users/jtiala/.nvm/versions/node/v24.14.1/bin/node
    npm: 11.11.0 - /Users/jtiala/.nvm/versions/node/v24.14.1/bin/npm
    pnpm: 10.33.0 - /Users/jtiala/.nvm/versions/node/v24.14.1/bin/pnpm
```

**Comment by marcalexiei** [comment]
Fix available in [v3.5.3](https://github.com/marcalexiei/eslint-plugin-zod/releases/tag/v3.5.3).

Thanks for the report!
